src/views/ContentReferenceWidget.vue

```diff
@@ -66,13 +67,98 @@ export default {
 		},
 	},
 
+	data() {
+		return {
+			element: {
+				...this.richObject,
+				isView: this.richObject.link.includes('view'),
+			},
+		}
+	},
+
+	computed: {
+		tableId() {
+			const id = this.richObject.link.split('/').at(-2)
+			return parseInt(id)
```

Comment:
We probably should just reuse the parsing that the backend already does. Currently we do not expose this but we can just use a patch like this to have the id and type available within the richObject

```diff
diff --git a/lib/Reference/ContentReferenceHelper.php b/lib/Reference/ContentReferenceHelper.php
index 610282c5..2f070d09 100644
--- a/lib/Reference/ContentReferenceHelper.php
+++ b/lib/Reference/ContentReferenceHelper.php
@@ -5,6 +5,7 @@ namespace OCA\Tables\Reference;
 use Exception;
 use OC\Collaboration\Reference\LinkReferenceProvider;
 use OCA\Tables\AppInfo\Application;
+use OCA\Tables\Db\Table;
 use OCA\Tables\Errors\InternalError;
 use OCA\Tables\Errors\NotFoundError;
 use OCA\Tables\Errors\PermissionError;
@@ -95,6 +96,9 @@ class ContentReferenceHelper extends ReferenceHelper {
                        $reference = new Reference($referenceText);
                        $referenceInfo = [];

+                       $referenceInfo['id'] = $elementId;
+                       $referenceInfo['type'] = $element instanceof Table ? Application::NODE_TYPE_TABLE : Application::NODE_TYPE_VIEW;
+
                        if ($element->getEmoji()) {
                                $reference->setDescription($element->getEmoji() . ' ' . $element->getTitle());
                                $referenceInfo['title'] = $element->getTitle();
```